scrape/scrape.go

```diff
@@ -617,6 +617,40 @@ func (sp *scrapePool) refreshTargetLimitErr() error {
 	return nil
 }
 
+func (sp *scrapePool) schemesFromConfig() (validationScheme model.ValidationScheme, escapingScheme model.EscapingScheme, err error) {
```

Comment:
As suggested in the previous comment, let's remove this function and have a ready `validationScheme model.ValidationScheme` and `escapingScheme model.EscapingScheme` passed to scrape structures. There's no need to do this down deep for every scrape loop.  Let's do it in the most earliest moment possible so default/validation in config layer and converting to model (if we need) on ApplyConfig in Scrape Manager.

Reply:
... and do we need conversions to model? Can't we just do it in config later to (the `model.ToEscapingScheme` part) 🤔

Reply:
it does seem like I could have used a string primitive for the schemes instead of integer enums, that's my C++ background showing through :).

Reply:
reload and also newScrapePool :)